### PR TITLE
Update JsPlugin for proper server-side support

### LINKED ISSUE #15: Finalize JsPlugin
JsPlugin was never used actually used server side. With this feature being re-prioritized, we'll probably want to take this opportunity to re-review and update JsPlugin as appropriate.

See https://github.com/deephaven/deephaven-core/pull/4925
See https://github.com/deephaven/deephaven-plugins/pull/150

### PR DESCRIPTION
**Pull request by devinrsmith**
In addition, updates JsPlugin documentation to be consistent with https://github.com/deephaven/deephaven-core/pull/4925

Fixes #15

### CODE REVIEW

File: src/deephaven/plugin/js.py [hunk 1]
```diff
@@ -6,29 +6,44 @@
...
+    construction for JS plugins.
+    """
 
     @abc.abstractmethod
-    def distribution_path(self) -> typing.Generator[pathlib.Path, None, None]:
-        """A generator that yields the distribution path directory."""
+    def root(self) -> typing.Generator[pathlib.Path, None, None]:
+        """
+        The root directory of the resources to serve. The root must exist.
+        """
```
Comment (mofojed) [inline src/deephaven/plugin/js.py]: Any reason to prefer `root` over `path`? Feel like `def path` would be the more natural name.

Comment (devinrsmith) [reply src/deephaven/plugin/js.py]: I don't have a strong opinion. `root`, `path`, `root_path` all seem reasonable to me. @niloc132? Either way, I think we should mirror this on the java side; `root`, `path`, or `rootPath`.

Comment (devinrsmith) [reply src/deephaven/plugin/js.py]: My only pushback against `path` would be that "path" is already implied by the return type.

**Review by mofojed — APPROVED** [review]
Looks good to me, some comments on the other ticket: https://github.com/deephaven/deephaven-plugins/pull/150/files

File: src/deephaven/plugin/js.py (hunk 1 above)
Comment (niloc132) [reply src/deephaven/plugin/js.py]: I'd lean toward (but not block on) making path part of the name yeah - even in Java I think I'd prefer that, but doubly so in python where types not even strong suggestions...

Comment (devinrsmith) [reply src/deephaven/plugin/js.py]: Will do.

File: src/deephaven/plugin/js.py [hunk 2]
```diff
@@ -6,29 +6,44 @@
...
+    """
+    A JS plugin is a Plugin that allows adding javascript code under the server's URL path "js-plugins/".
+    See https://github.com/deephaven/deephaven-plugins#js-plugins for more details about the underlying
+    construction for JS plugins.
+    """
 
     @abc.abstractmethod
-    def distribution_path(self) -> typing.Generator[pathlib.Path, None, None]:
-        """A generator that yields the distribution path directory."""
+    def root_path(self) -> typing.ContextManager[pathlib.Path]:
```
Comment (mofojed) [inline src/deephaven/plugin/js.py]: I still would prefer just `path` ... `root_path` seems redundant.

**Review by mofojed — COMMENTED** [review]
I'm alright with it, `root_path` just seems redundant.

**Review by niloc132 — APPROVED** [review]
Approved, one clarification: the intention w.r.t. allow/deny lists for files in `path` is to add a new, non-abstract method in the future as needed?

**Comment by devinrsmith** [comment]
> Approved, one clarification: the intention w.r.t. allow/deny lists for files in `path` is to add a new, non-abstract method in the future as needed?

If needed, yes, something like that. Or regex, glob, etc.

**Pull request merged**